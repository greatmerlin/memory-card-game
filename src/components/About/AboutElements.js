import styled from "styled-components";
import { Link as LinkRouter } from "react-router-dom";

export const Container = styled.div`
  min-height: 200px;
  padding: 5em;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  z-index: 0;
  overflow: hidden;
  background: linear-gradient(
    108deg,
    rgba(1, 147, 86, 1) 0%,
    rgba(10, 201, 122, 1) 100%
  );
`;

export const FormWrap = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  @media screen and (max-width: 400px) {
    height: 80%;
  }
`;

export const Icon = styled(LinkRouter)`
  margin-left: 32px;
  margin-top: 32px;
  text-decoration: none;
  color: #fff;
  font-weight: 700;
  font-size: 32px;
  padding: 1em;

  @media screen and (max-width: 480px) {
    margin-left: 16px;
    margin-top: 8px;
  }
`;

export const FormContent = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  @media screen and (max-width: 480px) {
    padding: 10px;
  }
`;

export const BlackContainer = styled.div`
  background: #010101;
  max-width: 400px;
  height: auto;
  width: 100%;
  display: grid;
  margin: 0 auto;
  z-index: 1;
  padding: 80px 32px;
  border-radius: 4px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.9);

  @media screen and (max-width: 400px) {
    padding: 32px 32px;
  }
`;

export const FormH1 = styled.h1`
  margin-bottom: 40px;
  color: #fff;
  font-size: 20px;
  font-weight: 400;
  text-align: center;
`;

export const BackButton = styled(LinkRouter)`
  background: #000;
  padding: 10px;
  border-radius: 4px;
  color: #01bf71;
  font-size: 20px;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  margin: 10px;
`;

export const Text = styled.span`
  text-align: center;
  margin-top: 24px;
  color: #fff;
  font-size: 14px;
`;
