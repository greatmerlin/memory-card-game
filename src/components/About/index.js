import React, { useState } from "react";
import Sidebar from "../Sidebar";
import SilentNavbar from "../SilentNavbar/index.js";
import Footer from "../Footer";

import {
  Container,
  FormWrap,
  Icon,
  FormContent,
  BlackContainer,
  Text,
  BackButton,
} from "./AboutElements.js";

function About() {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => {
    setIsOpen(!isOpen);
    console.log("ISOPEN---> ", isOpen);
  };
  return (
    <>
      <Sidebar isOpen={isOpen} toggle={toggle} />
      <SilentNavbar toggle={toggle} />
      <Container>
        <FormWrap>
          <Icon>About this game</Icon>
          <FormContent>
            <BlackContainer>
              <Text>
                Memory UP is a card game in which all cards are laid face down
                on a surface and two cards are flipped face up over each turn.
                The object of the game is to turn over pairs of matching cards.
                However, this is a special version of the game. The difference
                is that each player gets some coins, depending on the difficulty
                he selects. If he doen't find a matching pair, he loses a coin.
                The game finishes as soon as the player has lost all of his
                coins.
                <br />
                This game was developed from "Theologos Baxevanos" as part of a
                semester exercise for the Fernfachhochschule Schweiz, university
                of applied sciences in Switzerland.
                <br />
                The project could not have been finished without the support of
                the advanced web technologies lecturer "Philipp Lauwiner". A
                huge thank you for making this subject fun and interesting to
                every single one of us.
                <br />
                Have fun playing the game! :)
              </Text>
              ÃŽ
            </BlackContainer>
          </FormContent>
        </FormWrap>
        <br />
        <BackButton to="/"> Back </BackButton>
      </Container>
      <Footer />
    </>
  );
}

export default About;
