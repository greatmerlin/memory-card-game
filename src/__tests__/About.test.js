import React from "react";
import About from "../components/About/index.js";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';
import { Icon } from "../components/About/AboutElements.js";

configure({ adapter: new Adapter() });

// Testing area

it("renders About page without crashing", () => {
  shallow(<About />);
});

it("renders About header", () => {
  const wrapper = shallow(<About />);
  const welcome = <Icon>About this game</Icon>;
  expect(wrapper.contains(welcome)).toEqual(true);
});

it("about component renders correctly", () => {
  const tree = shallow(<About />);
  expect(toJson(tree)).toMatchSnapshot();
});