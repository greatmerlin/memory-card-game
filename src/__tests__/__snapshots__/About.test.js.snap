// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about component renders correctly 1`] = `
<Fragment>
  <Sidebar
    isOpen={false}
    toggle={[Function]}
  />
  <SilentNavbar
    toggle={[Function]}
  />
  <styled.div>
    <styled.div>
      <Styled(Link)>
        About this game
      </Styled(Link)>
      <styled.div>
        <styled.div>
          <styled.span>
            Memory UP is a card game in which all cards are laid face down on a surface and two cards are flipped face up over each turn. The object of the game is to turn over pairs of matching cards. However, this is a special version of the game. The difference is that each player gets some coins, depending on the difficulty he selects. If he doen't find a matching pair, he loses a coin. The game finishes as soon as the player has lost all of his coins.
            <br />
            This game was developed from "Theologos Baxevanos" as part of a semester exercise for the Fernfachhochschule Schweiz, university of applied sciences in Switzerland.
            <br />
            The project could not have been finished without the support of the advanced web technologies lecturer "Philipp Lauwiner". A huge thank you for making this subject fun and interesting to every single one of us.
            <br />
            Have fun playing the game! :)
          </styled.span>
          ÃŽ
        </styled.div>
      </styled.div>
    </styled.div>
    <br />
    <Styled(Link)
      to="/"
    >
       Back 
    </Styled(Link)>
  </styled.div>
  <Footer />
</Fragment>
`;
